pkg/server/server_test.go

```diff
@@ -505,6 +505,71 @@ func TestAvoidDeadlockWithinSingleCheckRequest(t *testing.T) {
 	require.False(t, resp.GetAllowed())
 }
 
+func TestRequestContextPropagation(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	for _, tc := range []struct {
+		name                   string
+		shouldPropagateContext bool
+	}{
+		{name: "disabled", shouldPropagateContext: false},
+		{name: "enabled", shouldPropagateContext: true},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			storeID := ulid.Make().String()
+			modelID := ulid.Make().String()
+
+			mockController := gomock.NewController(t)
+			t.Cleanup(mockController.Finish)
+
+			parentCtx, cancelParentCtx := context.WithCancel(context.Background())
+			t.Cleanup(cancelParentCtx)
+
+			mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)
+
+			mockDatastore.EXPECT().
+				ReadAuthorizationModel(gomock.Any(), storeID, modelID).
+				AnyTimes().
+				Return(&openfgav1.AuthorizationModel{}, nil)
+
+			mockDatastore.EXPECT().
+				ReadAuthorizationModel(gomock.Any(), gomock.Eq(storeID), gomock.Eq(modelID)).
+				AnyTimes().
+				DoAndReturn(func(ctx context.Context, _, _ string) (*openfgav1.AuthorizationModel, error) {
+					cancelParentCtx()
+					if tc.shouldPropagateContext {
+						require.ErrorIs(t, ctx.Err(), context.Canceled, "storage context must get cancelled if the request context is")
+					} else {
+						e := ctx.Err()
+						require.NoError(t, e, "storage context must not get canceled if request context is")
+					}
```

Comment:
```suggestion
					e := ctx.Err() 
					if tc.shouldPropagateContext {
						require.ErrorIs(t, e, context.Canceled, "storage context must get cancelled if the request context is propagate context")
					} else {
						require.NoError(t, e, "storage context must not get canceled if request context is is disable propagate context")
					}
```
nitpik: moving the e out of individual conditional branch makes it easier to read.